Parser/pegen/pegen.c

```diff
@@ -526,12 +526,7 @@ Parser_New(struct tok_state *tok, mod_ty (*parse_func)(Parser *), int input_mode
     p->size = 1;
 
     p->arena = arena;
-
-    if (fill_token(p) < 0) {
```

Comment:
Why are we removing this now? I cannot see what piece of the code now does this instead. Maybe it was not needed?

Reply:
> I cannot see what piece of the code now does this instead. Maybe it was not needed?

I was indeed not needed, but wasn't hurting either. It wasn't needed, because it gets called by `expect_token`, which is always called by the parser itself.

> Why are we removing this now?

Because it hurts now. At the time this is called here, keywords are not initialized yet, which means that if the first token is a keyword it parsed as a NAME, thus allowing things like `lambda = 3`.

Reply:
Gotcha! 👍